executor/executor.go

```diff
@@ -130,14 +137,57 @@ func (e *baseExecutor) Next(ctx context.Context, req *chunk.RecordBatch) error {
 	return nil
 }
 
-func newBaseExecutor(ctx sessionctx.Context, schema *expression.Schema, id string, children ...Executor) baseExecutor {
+// hash returns the signature of the executor
+func (e *baseExecutor) getHash() []byte {
+	return computeHashImpl(e.hashBuf, e.id, e.noChunksUsed)
+}
+
+func computeHashImpl(hashBuf []byte, id string, noChunksUsed int) []byte {
+	var (
+		eid        = hack.Slice(id)
+		bufferSize = len(eid) + 1*8
+	)
+	if cap(hashBuf) < bufferSize {
+		hashBuf = make([]byte, bufferSize)
+	}
+	hashBuf = hashBuf[:0]
+	hashBuf = append(hashBuf, eid...)
+	hashBuf = codec.EncodeInt(hashBuf, int64(noChunksUsed))
+	return hashBuf
+}
+
+// reuseChunkImpl returns a new chunk or an used one to buffer current executor's result.
+func reuseChunkImpl(ctx sessionctx.Context, planKey []byte, hash []byte, rt []*types.FieldType, cap int, sz int) *chunk.Chunk {
+	if len(planKey) > 0 {
+		key := plannercore.ConvertPlanCacheKey(planKey)
+		if val, hit := ctx.PreparedPlanCache().Get(key); hit {
+			chks := val.(*plannercore.PSTMTPlanCacheValue).Chunks
+			if chk, ok := chks[string(hash)]; ok {
```

Comment:
Thank you for the detailed comment. The updated PR should fix the race-condition because it introduces a new sequence number in order to differentiate the workers.

Reply:
I'm trying another way to resolve the issue instead of introducing the new sequence number.

Reply:
The updated PR extends  the executor id by augmenting the worker id to identify each executor uniquely:
https://github.com/pingcap/tidb/blob/b89108bb479e2bfe1b316bbe786cb65e704033ee/executor/distsql.go#L526